djangosaml2idp/views.py

```diff
@@ -143,16 +145,26 @@ def render_login_html_to_string(self, context=None, request=None, using=None):
         """ Render the html response for the login action. Can be using a custom html template if set on the view. """
         default_login_template_name = 'djangosaml2idp/login.html'
         custom_login_template_name = getattr(self, 'login_html_template', None)
+
         if custom_login_template_name:
-            try:
-                template = get_template(custom_login_template_name, using=using)
-            except (TemplateDoesNotExist, TemplateSyntaxError) as e:
-                logger.error('Specified template {} cannot be used due to: {}. Falling back to default login template {}'.format(custom_login_template_name, str(e), default_login_template_name))
-                template = get_template(default_login_template_name, using=using)
-        else:
-            template = get_template(default_login_template_name, using=using)
+            template = self._fetch_custom_template(custom_login_template_name, default_login_template_name, using)
+            return template.render(context, request)
+
+        template = get_template(default_login_template_name, using=using)
         return template.render(context, request)
```

Comment:
There was no test coverage for the `render_login_html_to_string` method so I added some and then refactored for clarity.